### PR TITLE
[unit-tests] added more unit tests for PublicFiles view

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
- added unit tests for `PublicFiles` view

## Related Issue

- #5237 

## How Has This Been Tested?

- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-app-files/tests/unit/views/PublicFiles.spec.ts [hunk 1]
```diff
@@ -1,17 +1,46 @@
...
+  'no-content-message': true,
+  'resource-table': true,
+  'context-actions': true,
+  pagination: true,
+  'list-info': true,
+  'router-link': RouterLinkStub
+}
+
+const $route = {
+  name: 'files-personal',
```
Comment (pascalwengerter) [inline packages/web-app-files/tests/unit/views/PublicFiles.spec.ts:23]: This shouldn't be the route name for public files AFAIK :)

Comment (kiranparajuli589) [reply packages/web-app-files/tests/unit/views/PublicFiles.spec.ts]: fixed :)

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPermission</strong> https://drone.owncloud.com/owncloud/web/21302/35/1
The following scenarios passed on retry:
- webUISharingPermissionsUsers/sharePermissionsUsers.feature:121

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingExternalRoot</strong> https://drone.owncloud.com/owncloud/web/21317/41/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

File: packages/web-app-files/tests/unit/views/PublicFiles.spec.ts [hunk 2]
```diff
@@ -20,4 +49,214 @@ describe('PublicFiles view', () => {
...
+      const selectedResourceList = [resourceList[0], resourceList[1]]
+      const notSelectedResourceList = [resourceList[2]]
+
+      it('should load the resource table with the correct props', () => {
+        const wrapper = getMountedWrapper({
+          paginatedResources: resourceList,
+          currentFolder: createFolder()
+        })
+        const resourceTableProps = wrapper.find(stubSelectors.resourceTable).props()
+        expect(resourceTableProps).toMatchObject({
```
Comment (kulmann) [inline packages/web-app-files/tests/unit/views/PublicFiles.spec.ts:125]: Please make use snapshot testing here. Changing the object in this test whenever something changes in the code is quite cumbersome.

**Review by kulmann — CHANGES_REQUESTED** [review]

File: packages/web-app-files/tests/unit/views/PublicFiles.spec.ts [hunk 3]
```diff
@@ -20,4 +49,214 @@ describe('PublicFiles view', () => {
...
+      it('should be visible if the length of the paginated resources is zero', () => {
+        const wrapper = getMountedWrapper({
+          paginatedResources: [],
+          currentFolder: {
+            canCreate: jest.fn()
+          }
+        })
+        expect(wrapper.find(stubSelectors.noContentMessage).exists()).toBeTruthy()
+      })
+      it('should show the "call to action" information if the currentFolder has "canCreate" permission', () => {
```
[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply packages/web-app-files/tests/unit/views/PublicFiles.spec.ts]: fixed

File: packages/web-app-files/tests/unit/views/PublicFiles.spec.ts (hunk 2 above)
Comment (kiranparajuli589) [reply packages/web-app-files/tests/unit/views/PublicFiles.spec.ts]: fixed

**Comment by kiranparajuli589** [comment]
@kulmann thx, I addressed your reviews. please review again!

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic1</strong> https://drone.owncloud.com/owncloud/web/21519/64/1
The following scenarios passed on retry:
- webUISharingPublicExpire/shareByPublicLinkExpiringLinks.feature:11

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp1</strong> https://drone.owncloud.com/owncloud/web/21900/68/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by kiranparajuli589** [comment]
build restarted

**Review by pascalwengerter — APPROVED** [review]
LGTM now, if @kulmann approves I'm happy to merge!

**Comment by phil-davis** [comment]
This PR was already 78 commits behind master. I rebased it to get fresh CI so that we are sure that the unit tests pass with current master.

@kulmann please review.

**Review by kulmann — APPROVED** [review]

**Pull request merged**